File: level3/풍선터트리기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> a) {
    int a_size = a.size();
    vector<bool> check(a_size,false);
    
    int a_min = a[0];
    int answer = a_size;
    for(int i = 1; i < a_size-1; i++){
        if(a_min < a[i]){
            check[i] = true;
        }
        else{
            a_min = a[i];
        }
    }
    
    a_min = a[a_size-1];
    for(int i = a_size-2; i > 0; i--){
        if(a_min < a[i]){
            if(check[i]){
                answer--;
            }
        }
        else{
            a_min = a[i];
        }
    }
    return answer;
}

File: level3/풍선터트리기.py
def solution(a):
    answer = min(2,len(a))
    check = [False]*len(a)
    
    left = float('inf')
    for i in range(1,len(a)-1):
        left = min(left, a[i-1])
        if left > a[i]:
            check[i] = True
            answer += 1
            
    right = float('inf')
    for i in range(len(a)-2,0,-1):
        right = min(right,a[i+1])
        if right > a[i] and not check[i]:
            answer += 1
        
    return answer
